crates/adapters/src/transport/http/input.rs

```diff
@@ -157,11 +166,7 @@ impl HttpInputEndpoint {
                         Err(_elapsed) => (),
                         Ok(Some(Ok(bytes))) => {
                             num_bytes += bytes.len();
-                            let mut new_errors = self.push_bytes(&bytes);
-                            num_errors += new_errors.len();
-                            for error in new_errors.drain(..) {
-                                errors.push(error);
-                            }
+                            num_errors += self.push(Some(&bytes), &mut errors);
```

Comment:
You changed the documentation of `start`/`pause` methods forbidding the connector to push data after a pause, but here it's possible that the controller calls pause between the status check above and this line, which will push updates to the circuit.

Reply:
Fixed because I made http queue the same way as the others.